### PR TITLE
16596: Remove additional services automapper profiles

### PR DESCRIPTION
**Pull request by PhysicsJon**

### CODE REVIEW

**Review by bjss-colin — DISMISSED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs [hunk 1]
```diff
@@ -107,11 +108,26 @@ public async Task<IActionResult> AssociatedServices(CatalogueItemId solutionId)
...
+            var additionalServices = item.Supplier.CatalogueItems
+                .Where(ci => ci.CatalogueItemType == CatalogueItemType.AdditionalService)
+                .OrderBy(ci => ci.Name)
+                .Select(ci => new AdditionalServiceModel
+                {
+                    Name = ci.Name,
+                    Description = ci.AdditionalService.FullDescription,
+                    Prices = ci.CataloguePrices.Select(cp => cp.ToString()).ToList(),
+                    SolutionId = ci.Id,
+                })
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs:124]: Worth adding a constructor to `AdditionalServiceModel` that takes a `CatalogueItem` and doing the initialization there?

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Models/CataloguePrice.cs [hunk 2]
```diff
@@ -35,5 +35,10 @@ public CataloguePrice()
         public TimeUnit? TimeUnit { get; set; }
 
         public ICollection<CataloguePriceTier> CataloguePriceTiers { get; set; }
+
+        public override string ToString()
+        {
+            return $"£{Price.Value:F} {PricingUnit?.Description}".Trim();
+        }
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Models/CataloguePrice.cs:42]: Good idea. I should have done this. 😃 

Can you update the `AssociatedServiceModel` constructor to use `ToString` instead so we don't have the same logic duplicated?

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs [hunk 3]
```diff
@@ -107,11 +108,26 @@ public async Task<IActionResult> AssociatedServices(CatalogueItemId solutionId)
...
+                .ToList();
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs:125]: Worth moving to the `AdditionalServicesModel` constructor?

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Models/CataloguePrice.cs (hunk 2 above)
Comment (PhysicsJon) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Models/CataloguePrice.cs:42]: done

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs (hunk 1 above)
Comment (PhysicsJon) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs:124]: done

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs (hunk 3 above)
Comment (PhysicsJon) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Solutions/Controllers/SolutionsController.cs:125]: done

**Review by bjss-chris — APPROVED** [review]

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**